util/dvsim/FlowCfg.py

```diff
@@ -411,36 +411,43 @@ def gen_results(self, results):
 
         '''
         for item in self.cfgs:
+            json_result = item._gen_json_results(results)
             result = item._gen_results(results)
             log.info("[results]: [%s]:\n%s\n", item.name, result)
             log.info("[scratch_path]: [%s] [%s]", item.name, item.scratch_path)
-            item.write_results_html(self.results_html_name, item.results_md)
+            item.write_results(self.results_html_name, item.results_md,
+                               json_result)
             log.log(VERBOSE, "[report]: [%s] [%s/report.html]", item.name,
                     item.results_dir)
             self.errors_seen |= item.errors_seen
 
         if self.is_primary_cfg:
             self.gen_results_summary()
-            self.write_results_html(self.results_html_name,
-                                    self.results_summary_md)
+            self.write_results(self.results_html_name,
+                               self.results_summary_md)
 
     def gen_results_summary(self):
         '''Public facing API to generate summary results for each IP/cfg file
         '''
         return
 
-    def write_results_html(self, filename, text_md):
+    def write_results(self, html_filename, text_md, json_str=None):
         """Converts md text to HTML and writes to results_dir area."""
```

Comment:
Do you mind also updating the comment to reflect the change?

Reply:
Good catch, thanks! Fixed